src/test/java/cat/udl/eps/softarch/academicrecruit/steps/DeleteCandidateStepDefs.java

```diff
@@ -0,0 +1,54 @@
+package cat.udl.eps.softarch.academicrecruit.steps;
+
+        import cat.udl.eps.softarch.academicrecruit.domain.Candidate;
+        import cat.udl.eps.softarch.academicrecruit.domain.Participant;
+        import cat.udl.eps.softarch.academicrecruit.repository.CandidateRepository;
+        import cat.udl.eps.softarch.academicrecruit.repository.ParticipantRepository;
+        import io.cucumber.java.en.And;
+        import io.cucumber.java.en.When;
+        import org.springframework.http.MediaType;
+
+        import static org.hamcrest.Matchers.is;
+        import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+        import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+        import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+        import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+
+public class DeleteCandidateStepDefs {
+    final StepDefs stepDefs;
+    final CandidateRepository candidateRepository;
+    private String newResourceUri;
+
+    public DeleteCandidateStepDefs(StepDefs stepDefs, CandidateRepository candidateRepository) {
+        this.stepDefs = stepDefs;
+        this.candidateRepository = candidateRepository;
+    }
+
+
+    @When("I delete a candidate with name {string}")
+    public void iDeleteCandidateTo(String name) throws Throwable {
+
+        Candidate candidate = new Candidate();
```

Comment:
It is better to create the candidate in a separate @Given step like `There exist a candidate with name...` and then here recover the candidate using a findByName and then its URI using the .getUri() method